fileServer.js

```diff
@@ -0,0 +1,96 @@
+ // 'use strict';
+
+ var http = require('http');
+ var fs = require('fs');
+ var path = require('path');
+ var url = require('url');
+ var rootDir = __dirname + '/public';
+
+ const mimeType = {
+   'html': 'text/html',
+   'jpeg': 'image/jpeg',
+   'jpg': 'image/jpeg',
+   'png': 'image/png',
+   'js': 'text/javascript',
+   'css': 'text/css'
+ };
+
+ function generate500error(response, errorMessage) {
+   response.writeHead(500, {
+     "Content-Type": "text/html"
+   });
+
+   response.write(`<h1>Server cannot process the request: ${errorMessage}</h1>`);
+   // Log the error message for the opearations people
+   console.log(`Server cannot process the request: ${errorMessage}`);
+   // Show which function called us and which file and LINE the call was made from
+   console.trace("caller is " + generate500error.caller.toString());
```

Comment:
What we want to do to make your code production-ready is the line where the caller function was called on and the file that function is in ... the source code in a production setting would be considered a security risk.

Intentionally create an error in your server so that when a route is called it throws a syntax error, look at how that stack trace is formatted.

node.js allows developers to access a stack trace themselves using the Error object. This stackoverflow post clued me into that direction:

https://stackoverflow.com/questions/16697791/nodejs-get-filename-of-caller-function/29581862#29581862

https://nodejs.org/api/errors.html#errors_class_error

Read up on the stack property and captureStackTrace() and then give this another shot.